src/rebar_dir.erl

```diff
@@ -103,12 +103,28 @@ home_dir() ->
     {ok, [[Home]]} = init:get_argument(home),
     Home.
 
+
 %% @doc returns the directory where the global configuration files for rebar3
 %% may be stored.
 -spec global_config_dir(rebar_state:t()) -> file:filename_all().
 global_config_dir(State) ->
-    Home = home_dir(),
-    rebar_state:get(State, global_rebar_dir, filename:join([Home, ".config", "rebar3"])).
+    filename:join([rebar_config_dir(State), ".config", "rebar3"]).
+    %% REVIEW:
+    %% This negates the line below because one of rebar_config_dir, REBAR_CONFIG_DIR, or HOME will always be true.
+    %% Is there a case where this will have negative side effects? In other words, going forward, it seems
+    %% we always want $HOME or global_rebar_dir to be the default, but which one?
+    %% rebar_state:get(State, global_rebar_dir, filename:join([rebar_config_dir(State), ".config", "rebar3"])).
```

Comment:
unless I'm wrong, this used to let people set `global_rebar_dir` such that if I used `/tmp/rebar3` the data would be directly there. This _now_ makes it go to `/tmp/rebar3/.config/rebar3` instead without having changed the setting.

That's a breaking change, but I don't know if it's significant enough to be a problem. For sure someone relying on it for builds will start having projects explode.

Also, refresh my memory why `global_rebar_dir` is deprecated to become `rebar_config_dir` ?

Reply:
I guess they're acceptable changes since both could be set at once and thus prevent clashes...

Reply:
If you look at the issue that sparked this PR you can see myself and OP are trying to make things work via REBAR_CACHE_DIR and $HOME, but I do not believe we tried using global_rebar_dir. However, when trying to use it as suggested I can not get anything to work. 

 Also,  I believe, I wasn't entirely sure what this would end up looking like and what would be done with `global_rebar_dir` in regards to config and this is PR was somewhat exploratory; once again based on https://github.com/tsloughter/rebar3_hex/issues/113

Reply:
Got a chance to glance at this. Yeah, I think there was just a boat load of confusion on my end, I'm not sure this PR is even needed. It seems like REBAR_GLOBAL_DIR (rebar_global_dir) does what we need. I think this option is not documented at least on the site, so maybe that's why OP on the other issue didn't know to use it. Likewise, I suppose in a hacking frenzy I misunderstand the option. 

 However and double check me on this, it seems as though you can only specify this in your rebar.config and not via an ENV var. If that's true, I could adjust this PR to do just that.